secure_storage: its: store: settings: allow using custom setting names

tomi-font opened the pull request:
Allow replacing the default naming scheme of the stored settings by providing a custom function that fills a name buffer based on the ITS entry UID.

tomi-font committed:
secure_storage: its: store: settings: allow using custom setting names

Allow replacing the default naming scheme of the stored settings by
providing a custom function that fills a name buffer based on the ITS
entry UID.

Signed-off-by: Tomi Fontanilles <tomi.fontanilles@nordicsemi.no>

ceolin commented on subsys/secure_storage/Kconfig.its_store:
```diff
@@ -60,8 +60,23 @@ endif # SECURE_STORAGE_ITS_STORE_IMPLEMENTATION_ZMS
+	bool "Custom naming scheme for the stored settings"
+	help
+	  Implement the secure_storage_its_store_settings_get_name() function
+	  declared in <zephyr/secure_storage/its/store/settings_get.h>
```
I am not a big fan of putting function /file names / paths. This can change and gets inconsistent. Besides that this should describe the functionality. Just say for what this symbol is used.

ceolin commented on subsys/secure_storage/include/internal/zephyr/secure_storage/its/store/settings_get.h:
```diff
@@ -0,0 +1,29 @@
+#include <zephyr/secure_storage/its/common.h>
+
+enum { SECURE_STORAGE_ITS_STORE_SETTINGS_NAME_BUF_SIZE
+	= CONFIG_SECURE_STORAGE_ITS_STORE_SETTINGS_NAME_MAX_LEN + 1 };
```
Curious about this construction. Why don't you simply do `#define SECURE_STORAGE_ITS_STORE_SETTINGS_NAME_BUF_SIZE (CONFIG_SECURE_STORAGE_ITS_STORE_SETTINGS_NAME_MAX_LEN + 1)`

valeriosetti commented on subsys/secure_storage/Kconfig.its_store:
```diff
@@ -60,8 +60,25 @@ endif # SECURE_STORAGE_ITS_STORE_IMPLEMENTATION_ZMS
+	int "Maximum setting name length"
+	range 2 64
+	default 22 if !SECURE_STORAGE_ITS_STORE_SETTINGS_NAME_CUSTOM
+	default 0
```
I saw that in `settings.c` you added checks for `SECURE_STORAGE_ITS_STORE_SETTINGS_NAME_MAX_LEN`, but that's only when `SECURE_STORAGE_ITS_STORE_SETTINGS_NAME_CUSTOM` is not defined, of course, because in that case all is pre-defined.

I like the idea to set `default 0` when `SECURE_STORAGE_ITS_STORE_SETTINGS_NAME_CUSTOM` is enabled so that the user is "forced" to specify a value for `SECURE_STORAGE_ITS_STORE_SETTINGS_NAME_MAX_LEN`. However there's no failure at build time that can warn the user about this. What about adding a check in the CMake file for this purpose?

tomi-font replied:
There actually is a build failure caused by this, that's exactly the purpose of having a default outside of the range:
```
warning: default value 0 on SECURE_STORAGE_ITS_STORE_SETTINGS_NAME_MAX_LEN (defined at subsys/secure_storage/Kconfig.its_store:78) clamped to 2 due to being outside the active range ([2, 64])
[...]
error: Aborting due to Kconfig warnings
```

valeriosetti replied:
Ops, you're absolutely right, sorry for missing that part. Please ignore my comment

valeriosetti reviewed (changes requested):
I only left one possible minor suggestion for improvement, but a part from that it looks OK to me

tomi-font committed:
secure_storage: fix formatting of CMake ifs

The rule says there must not be spaces "between a command and the opening
parenthesis".

Signed-off-by: Tomi Fontanilles <tomi.fontanilles@nordicsemi.no>

tomi-font commented on subsys/secure_storage/CMakeLists.txt:
```diff
@@ -46,3 +46,7 @@ if (CONFIG_SECURE_STORAGE_ITS_TRANSFORM_AEAD_SCHEME_CUSTOM
   make_available(its/transform/aead_get.h)
 endif()
+
+if (CONFIG_SECURE_STORAGE_ITS_STORE_SETTINGS_NAME_CUSTOM)
```
Fixed it as well as all the other occurrences in a separate commit.

ozersa reviewed (approved):
looks good to me.
